pkg/rtc/signalhandler.go

```diff
@@ -75,7 +75,7 @@ func HandleParticipantSignal(room types.Room, participant types.LocalParticipant
 		}
 	case *livekit.SignalRequest_Leave:
 		pLogger.Infow("client leaving room")
-		_ = participant.Close(true)
+		room.RemoveParticipant(participant.Identity())
```

Comment:
> But, `RoomManager` registers `OnClose` on `Participant` and stores room to update participant count. That would have had the correct participant count.

Do you mean it would not have had the correct participant count? I'm not following what the issue is.

Reply:
Sorry @davidzhao for the confusion. Yeah you are correct.

Room manager was registering on close on the participant. In that close handler, it was storing the room (mainly to upstate participant count).

Signal handler was calling participant close directly. Participants close sets the state to DISCONNECTED and fires a callback for participant update to room in a go routine. That handler in room.go fires another go routine to remove participant when state is DISCONNECTED. And remove participant is what reduces the participant count.

But in the meantime participant close is fired synchronously and room manager fields that and does a StoreRoom. So it may or may not have the correct participant count depending on when the other two go routines ran.